File: src/backend/src/main/java/br/com/grupopipa/gestaointegrada/financeiro/movimentacao/MovimentacaoFinanceiraDTO.java
package br.com.grupopipa.gestaointegrada.financeiro.movimentacao;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

import br.com.grupopipa.gestaointegrada.core.dto.DTO;

import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class MovimentacaoFinanceiraDTO implements DTO {

  private UUID id;
  private List<MovimentacaoTituloDTO> titulos;
  private UUID contaBancariaId;
  private String contaBancariaNome;
  private String tipo; // PAGAMENTO, RECEBIMENTO, ESTORNO, TRANSFERENCIA
  private String formaPagamento; // PIX, DINHEIRO, BOLETO, etc
  private BigDecimal valor;
  private LocalDate data;
  private String observacoes;
  private LocalDateTime createdAt;
  private LocalDateTime updatedAt;
  private String createdBy;
  private String updatedBy;
}

File: src/frontend/src/app/components/financeiro/movimentacao-financeira/model/movimentacao-financeira.dto.ts
export class MovimentacaoTituloDTO {
  id?: string;
  descricao?: string;
  valor?: number;
  tipo?: string;

  constructor(id?: string, descricao?: string, valor?: number, tipo?: string) {
    this.id = id;
    this.descricao = descricao;
    this.valor = valor;
    this.tipo = tipo;
  }
}

export class MovimentacaoFinanceiraDTO {
  id?: string;
  titulos: MovimentacaoTituloDTO[];
  contaBancariaId?: string;
  // legacy/display field kept for UI; backend expects contaBancariaId
  contaBancaria?: string;
  tipo: string;
  formaPagamento: string;
  valor: number;
  data: string;
  unidadeNegocioId: string;
  observacoes?: string;

  constructor(
    id?: string,
    titulos: MovimentacaoTituloDTO[] = [],
    contaBancariaId?: string,
    contaBancaria?: string,
    tipo = '',
    formaPagamento = '',
    valor = 0,
    data = '',
    unidadeNegocioId = '',
    observacoes?: string
  ) {
    this.id = id;
    this.titulos = titulos;
    this.contaBancariaId = contaBancariaId;
    this.contaBancaria = contaBancaria;
    this.tipo = tipo;
    this.formaPagamento = formaPagamento;
    this.valor = valor;
    this.data = data;
    this.unidadeNegocioId = unidadeNegocioId;
    this.observacoes = observacoes;
  }
}
